tests/integration/test_sns.py

```diff
@@ -2330,3 +2328,172 @@ def validate_content():
+            "AWS.SNS.MOBILE.APNS.TOPIC": {
+                "DataType": "String",
+                "StringValue": "com.amazon.mobile.messaging.myapp",
+            },
+            "AWS.SNS.MOBILE.APNS.PUSH_TYPE": {
+                "DataType": "String",
+                "StringValue": "background",
+            },
+            "AWS.SNS.MOBILE.APNS.PRIORITY": {
+                "DataType": "String",
+                "StringValue": "5",
+            },
+        }
+        # publish to a topic which has a platform subscribed to it
+        sns_client.publish(
+            TopicArn=topic_arn,
+            Message=message_for_topic,
+            MessageAttributes=message_attributes,
+            MessageStructure="json",
+        )
+        # publish directly to the platform endpoint
+        sns_client.publish(
+            TargetArn=endpoint_arn_2,
+            Message=message,
+            MessageAttributes=message_attributes,
+            MessageStructure="json",
+        )
+
+        # assert that message has been received
+        def check_message():
+            assert len(sns_backend.platform_endpoint_messages[endpoint_arn]) > 0
+
+        retry(check_message, retries=PUBLICATION_RETRIES, sleep=PUBLICATION_TIMEOUT)
+
+        msgs_url = config.get_edge_url() + INTERNAL_RESOURCE_PATH + PLATFORM_ENDPOINT_MSGS_ENDPOINT
+        api_contents = requests.get(msgs_url).json()
+
+        assert len(api_contents["platform_endpoint_messages"]) == 2
+        assert len(api_contents["platform_endpoint_messages"][endpoint_arn]) == 1
+        assert len(api_contents["platform_endpoint_messages"][endpoint_arn_2]) == 1
+        assert api_contents["region"] == "us-east-1"
+        # TODO: current implementation does not dispatch depending on platform type, we will have the message
+        # for all platforms
+        assert (
+            api_contents["platform_endpoint_messages"][endpoint_arn][0]["Message"]
+            == message_for_topic
+        )
+        assert (
+            api_contents["platform_endpoint_messages"][endpoint_arn][0]["MessageAttributes"]
+            == message_attributes
+        )
+
+        # Ensure messages can be filtered by EndpointArn
+        msg_with_endpoint = requests.get(msgs_url, params={"endpointArn": endpoint_arn}).json()
+        assert len(msg_with_endpoint["platform_endpoint_messages"]) == 1
+        assert len(msg_with_endpoint["platform_endpoint_messages"][endpoint_arn]) == 1
+        assert api_contents["region"] == "us-east-1"
+
+        # Ensure you can select the region
+        msg_with_region = requests.get(msgs_url, params={"region": "eu-west-1"}).json()
+        assert len(msg_with_region["platform_endpoint_messages"]) == 0
+        assert msg_with_region["region"] == "eu-west-1"
+
+        sns_client.delete_endpoint(EndpointArn=endpoint_arn)
+        sns_client.delete_endpoint(EndpointArn=endpoint_arn_2)
+        sns_client.delete_platform_application(PlatformApplicationArn=app_arn)
```

Comment:
Could we solve this via fixtures? So we can keep it in line with the rest of the resources.

Reply:
Yes, I'll create a fixture for platform_application and platform_endpoint, as we're using them more! Thanks!